src/slice_n_preprocess.py

```diff
@@ -28,39 +63,31 @@
 
     basin_path = base_dir / "vector_data/basin/Dudh_Koshi_Glacier.shp"
 
-    labels_path = base_dir / f'vector_data/{data_c["year"]}/{data_c["country"]}/data/Glacier_{data_c["year"]}.shp'
-    borders_path =  base_dir/ f'vector_data/borders/{data_c["country"]}/{data_c["country"]}.shp'
-    sat_dir = base_dir / f'img_data/{data_c["year"]}/{data_c["country"]}'
-    save_loc = base_dir / f'sat_files/{data_c["year"]}/{data_c["country"]}' 
+    countries = data_c["country"]
+    years = data_c["year"]
 
 
-    # slice all images in that folder
-    preprocess.chunck_sat_files(sat_dir, labels_path, base_dir, save_loc,
-                                borders_path=borders_path, basin_path=basin_path,
-                                size=(data_c["size"], data_c["size"]),
-                                year=data_c["year"], country=data_c["country"])
+    if not years:
+        years = [year_path.name for year_path in Path(base_dir, 'img_data').iterdir()]
 
-    def valid_cond_f(sat_data): return ((sat_data.labels_perc > valid_c["labels_perc"]) &
-                                        (sat_data.labels_in_border > valid_c["labels_in_border"]) &
-                                        (sat_data.is_nan_perc < valid_c["is_nan_perc"]))
+    if not countries:
+        countries = []
+        for year in years:
+            year_path = Path(base_dir, 'img_data', year)
+            for country_path in year_path.iterdir():
+                countries.append(country_path.name)
+        countries = set(countries)
     
-    sat_data_file = os.path.join(save_loc, 'sat_data.csv')
-    if split_c["random_test"]:
-        def test_cond_f(sat_data): return pd.Series([False for _ in range(len(sat_data))])
-        preprocess.filter_images(sat_data_file, valid_cond_f, test_cond_f)
-        preprocess.split_train_test(
-            sat_data_file, save=True, perc=split_c["test_perc"], label='test')
-
-    else:
-        def test_cond_f(sat_data): return (sat_data.basin_perc > 0)
-        preprocess.filter_images(sat_data_file, valid_cond_f, test_cond_f)
-
-    preprocess.split_train_test(sat_data_file, save=True, perc=split_c["dev_perc"])
-
-    if data_c["merge"]:
+    for year, country in list(itertools.product(years, countries)):
+        if Path(base_dir, f'img_data/{year}/{country}').exists():
+            logging.info(f'Processing {year}/{country}')
+            preprocess_country(base_dir, basin_path,
+                               country, year, data_c, valid_c, split_c)
+            
         sat_path = base_dir / "sat_files"
         dfs = []
         for year_path in sat_path.iterdir():
             for country_path in year_path.iterdir():
-                dfs.append(pd.read_csv(country_path / 'sat_data.csv'))
-        pd.concat(dfs).to_csv(base_dir / 'sat_data.csv')
\ No newline at end of file
+                if (country_path / 'sat_data.csv').exists():
```

Comment:
Can't you just use name_of_df.to_csv(..., mode="a") to append? Instead of reading in, concatenating, and then writing?

Reply:
That would be great, but it would result in repeated rows in case of running a file more than one. Which happens a lot, in development at least. This is the easiest way not to over complex the code and have some flexibility.

Reply:
Ah I see, didn't think about repeated data, this makes sense.